### PR TITLE
remove all expired sessions on getConnection

### PR DESCRIPTION
**Pull request by SimbaGithub**
salesforce 00399934, fix authentication token is expired issue.

### CODE REVIEW

**Comment by SimbaGithub** [comment]
salesforce 00399934, fix authentication token is expired issue.

File: Snowflake.Data/Client/SnowflakeDbConnectionPool.cs [hunk 1]
```diff
@@ -68,6 +68,7 @@ internal static SnowflakeDbConnection getConnection(string connStr)
             }
             lock (_connectionPoolLock)
             {
+                cleanExpiredConnections(); // clean up all expired connections first.
```
Comment (sfc-gh-igarish) [inline Snowflake.Data/Client/SnowflakeDbConnectionPool.cs:71]: Will this cause delay in getting a connection? If yes, then I suggest when connection string match in the pool, do one more check whether it's expired or not. If expired, then put in a separate list, say expire list and continue look into pool. Don't change pool count. After the loop, which are in expire list, remove those only. This way we remove expire connection in batches and not holding to get a connection from a pool to cleanup all expired connections.

File: Snowflake.Data/Client/SnowflakeDbConnectionPool.cs [hunk 2]
```diff
@@ -68,14 +68,25 @@ internal static SnowflakeDbConnection getConnection(string connStr)
...
                         connectionPool.RemoveAt(i);
-                        return conn;
+                        long timeNow = DateTimeOffset.Now.ToUnixTimeSeconds();
+                        if (conn._poolTimeout <= timeNow)
+                        {
+                            if (conn.SfSession != null)
+                            {
+                                conn.SfSession.close();
+                            }
+                            return null;
```
Comment (sfc-gh-igarish) [inline Snowflake.Data/Client/SnowflakeDbConnectionPool.cs:84]: Do we have only one connection in pool for matching connection string? If not, we need logic to continue and reduce i counter as we remove one element. Think about it.

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM. Once tests pass, you can merge.

**Pull request merged**